File: python/lld/board-games/snake-ladder/snake_ladder_move.py
from player import Player


# A record of one turn — also the replay/undo log entry.
class SnakeLadderMove:
    def __init__(self, player: Player, dice_value: int, from_cell: int, to_cell: int,
                 landed_on_snake: bool = False, landed_on_ladder: bool = False):
        self.player = player
        self.dice_value = dice_value
        self.from_cell = from_cell
        self.to_cell = to_cell
        self.landed_on_snake = landed_on_snake
        self.landed_on_ladder = landed_on_ladder

File: typescript/lld/board-games/snake-ladder/SnakeLadderMove.ts
import { Player } from './Player';

// A record of one turn — also the replay/undo log entry.
export interface SnakeLadderMove {
    player: Player;
    diceValue: number;
    fromCell: number;
    toCell: number;
    landedOnSnake?: boolean;
    landedOnLadder?: boolean;
}